ui/src/app/pages/login/account-creation/account-creation.tsx

```diff
@@ -0,0 +1,665 @@
+  TextArea,
+  TextInput,
+  ValidationError
+} from 'app/components/inputs';
+
+import {
+  TooltipTrigger
+} from 'app/components/popups';
+
+import {
+  profileApi
+} from 'app/services/swagger-fetch-clients';
+
+import {
+  DataAccessLevel,
+  Profile,
+} from 'generated/fetch/api';
+
+import colors from 'app/styles/colors';
+import {summarizeErrors} from 'app/utils/index';
+import {environment} from 'environments/environment';
+import * as fp from 'lodash/fp';
+import {Dropdown} from 'primereact/dropdown';
+import * as React from 'react';
+import * as validate from 'validate.js';
+
+function isBlank(s: string) {
+  return (!s || /^\s*$/.test(s));
+}
+
+// The values will change once we have ENUM as part of DB work as part of create account 2
+const Options = {
+  roles: [
+    {label: `Undergraduate (Bachelor level) student`, value: 'bachelor'},
+    {label: `Graduate trainee (Current student in a Masters, PhD, or Medical school training
+        program)`, value: 'trainee'},
+    {label: `Research fellow (a post-doctoral fellow or medical resident in training)`,
+      value: 'research'},
+    {label: `Early career tenure-track researcher`, value: 'earylCarrer'},
+    {label: `Non tenure-track researcher`, value: 'nontenure'},
+    {label: `Mid-career tenured researcher`, value: 'midCarrer'},
+    {label: `Late career tenured researcher`, value: 'lateCarrer'},
+    {label: `Project Personnel (eg: Research Assistant, Data Analyst, Project Manager, Research
+        Coordinator or other roles)`, value: 'project'}
+  ],
+  affiliations: [
+    {label: 'Industry', value: 'industry'},
+    {label: `Educational institution (High school, Community college, 4-year college, trade
+        school)`, value: 'educationalInstitution'},
+    {label: `Community Scientist (i.e. I am accessing AoU for independent research, unrelated to my
+        professional affiliation)`, value: 'scientist'},
+    {label: `Other (free text)`, value: 'freeText'}
+  ],
+  industryRole: [
+    {label: 'Research Assistant (pre-doctoral)', value: 'preDoctoral'},
+    {label: 'Research associate (post-doctoral; early/mid career)', value: 'postDoctoral'},
+    {label: 'Senior Researcher (PI/Team Lead)', value: 'teamLead'},
+    {label: 'Other (free text)', value: 'freeText'}
+  ],
+  eductionRole: [
+    {label: 'Teacher/Professor', value: 'teacher'},
+    {label: 'Student', value: 'student'},
+    {label: 'Administrator', value: 'admin'},
+    {label: 'Other (free text)', value: 'freeText'}
+  ]
+};
+
+// The following will be part of Profile object once create account survey is in
+export interface Address {
+  streetAddress1: string;
+  streetAddress2: string;
+  city: string;
+  state: string;
+  zipcode: string;
+  country: string;
+}
+
+export interface AccountCreationProps {
+  invitationKey: string;
+  setProfile: Function;
+}
+
+export interface AccountCreationState {
+  creatingAccount: boolean;
+  errors: any;
+  invalidEmail: boolean;
+  rolesOptions: any;
```

Comment:
I think this could be more descriptive than `any`. Like maybe: `Option[]`? (and then create an Option interface?)